salt/modules/yumpkg.py

```diff
@@ -3217,3 +3217,70 @@ def list_installed_patches():
         salt '*' pkg.list_installed_patches
     '''
     return _get_patches(installed_only=True)
+
+
+@salt.utils.decorators.path.which('yum-complete-transaction')
+def complete_transaction(cleanup_only=False, recursive=False, max_attempts=3):
+    '''
+    .. versionadded:: Fluorine
+
+    Execute ``yum-complete-transaction``, which is provided by the ``yum-utils`` package.
+
+    cleanup_only
+        Specify if the ``--cleanup-only`` option should be supplied.
+
+    recursive
+        Specify if ``yum-complete-transaction`` should be called recursively
+        (it only completes one transaction at a time).
+
+    max_attempts
+        If ``recursive`` is ``True``, the maximum times ``yum-complete-transaction`` should be called.
+
+    .. note::
+
+        Recursive calls will stop once ``No unfinished transactions left`` is in the returned output.
+
+    .. note::
+
+        ``yum-utils`` will already be installed on the minion if the package
+        was installed from the Fedora / EPEL repositories.
+
+    CLI example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.complete_transaction
+        salt '*' pkg.complete_transaction cleanup_only=True
+        salt '*' pkg.complete_transaction recursive=True max_attempts=5
+    '''
+
+    return _complete_transaction(cleanup_only, recursive, max_attempts, 1, '')
+
+
+def _complete_transaction(cleanup_only, recursive, max_attempts, run_count, ret):
+    '''
+    .. versionadded:: Fluorine
+
+    Called from ``complete_transaction`` to protect the arguments
+    used for tail recursion, ``run_count`` and ``ret``.
+    '''
+
+    cmd = ['yum-complete-transaction']
+    if cleanup_only:
+        cmd.append('--cleanup-only')
+    if ret:
+        # Ensure we have at least one trailing newline if this is not the first attempt
+        ret += '\n'
+    ret += __salt__['cmd.run'](
+        cmd,
+        output_loglevel='trace',
+        python_shell=False
+    )
+    if recursive and 'No unfinished transactions left' not in ret:
+        if run_count >= max_attempts:
+            ret += '\nAttempt {0}/{1} exceeded max'.format(run_count, max_attempts)
+            log.warning('Attempt %s/%s exceeded max for command: `%s`', run_count, max_attempts, ' '.join(cmd))
+        else:
+            return _complete_transaction(cleanup_only, recursive, max_attempts, run_count + 1, ret)
+
+    return ret
```

Comment:
I am not sure if this is a good idea to return string, deliberately losing the actual error code. This is all over the place for RH module though, which is even worse.